src/OIDCCryptUtils.purs

```diff
@@ -0,0 +1,48 @@
+module OIDCCryptUtils (RSASIGNTIME(), hashNonce, bindState, unbindState, verifyIdToken) where
```

Comment:
I personally don't think that type synonyms would be enough, because I can mix up arg order then. 
How records can be used here? 

Why don't you like newtypes? Using them doesn't change foreign code here.